org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/completion/LSContentAssistProcessor.java

```diff
@@ -341,6 +327,16 @@ private static IContextInformation toContextInformation(SignatureInformation inf
 		return new ContextInformation(information.getLabel(), signature.toString());
 	}
 
+	private void cancelFuture(@Nullable CompletableFuture<List<Void>> future) {
```

Comment:
how about adding it as `public static void cancelFuture(@Nullable Future<?> future)` to `org.eclipse.lsp4e.internal.CancellationUtil` and reuse it in other places too?

Reply:
Following up on your comment, I cannot find any other place in the codebase that uses this idiom. I have them read the documentation of `CompletableFuture.cancel(boolean)` and checked the implementation, and I think the exception will never be thrown. So we can just remove the catch block. 

Do you agree?

Reply:
Yes. Reading the JavaDoc and checking the implementation it looks like it is never thrown. So removing the try catch should be ok

Reply:
I still would suggest adding this as `public static void cancelFuture(@Nullable Future<?> future)` to `org.eclipse.lsp4e.internal.CancellationUtil` and reuse it in other places. There are a few more places where a future is first checked for null and then canceled and if we have a method here already why not use it everywhere else.

Reply:
I will take a look.